rust/agama-server/src/hostname/web.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) [2025] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+//! This module implements the web API for the hostname service.
+//!
+//! The module offers one public function:
+//!
+//! * `hostname_service` which returns the Axum service.
+//!
+//! stream is not needed, as we do not need to emit signals (for NOW).
+
+use agama_lib::{
+    error::ServiceError,
+    hostname::{client::HostnameClient, model::HostnameSettings},
+};
+use axum::{extract::State, routing::put, Json, Router};
+
+use crate::error;
+
+#[derive(Clone)]
+struct HostnameState<'a> {
+    client: HostnameClient<'a>,
+}
+
+/// Sets up and returns the axum service for the hostname module.
+pub async fn hostname_service() -> Result<Router, ServiceError> {
+    let dbus = zbus::Connection::system().await?;
+    let client = HostnameClient::new(dbus).await?;
```

Comment:
does client even need dbus parameter when it always use system bus? I think it at least deserve here explanation why it always use system bus instead of agama one passed as parameter like other services did

Reply:
we can ommit it, just modified the pass of the dbus parameter of the hostname_service but not the hostname client and as commented above used bootloader as skeleton, but as you commented in this case we do not need to pass the dbus zbus connnection.